ui/tests/acceptance/job-definition-test.js

```diff
@@ -130,3 +130,48 @@ module('Acceptance | job definition', function (hooks) {
     );
   });
 });
+
+module('display and edit using full specification', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+  setupCodeMirror(hooks);
+
+  hooks.beforeEach(async function () {
+    server.create('node');
+    server.create('job');
+    job = server.db.jobs[0];
+  });
+
+  test('it allows users to select between full specification and JSON definition', async function (assert) {
+    assert.expect(3);
+    const specification_response = {
+      Format: 'hcl2',
+      JobID: 'example',
+      JobIndex: 223,
+      Namespace: 'default',
+      Source:
+        'variable "datacenter" {\n  description = "The datacenter to run the job in"\n  type        = string\n  default     = "dc1"\n}\n\njob "example" {\n  datacenters = [var.datacenter]\n\n  group "example-group" {\n    task "example-task" {\n      driver = "docker"\n\n      config {\n        image = "redis:3.2"\n      }\n\n      resources {\n        cpu    = 500\n        memory = 256\n      }\n    }\n  }\n}\n',
+      VariableFlags: { datacenter: 'dc2' },
+      Variables: '',
+      Version: 0,
+    };
+    server.get('/job/:id', () => JOB_JSON);
+    server.get('/job/:id/submission', () => specification_response);
+
+    await Definition.visit({ id: job.id });
+
+    assert
+      .dom('[data-test-select="job-spec"]')
+      .exists('A select button exists and defaults to full definition');
+    let codeMirror = getCodeMirrorInstance('[data-test-editor]');
+    assert.equal(
+      codeMirror.getValue(),
+      specification_response.Source,
+      'Shows the full definition as written by the user'
+    );
+
+    await click('[data-test-select-full]');
+    codeMirror = getCodeMirrorInstance('[data-test-editor]');
+    assert.propContains(JSON.parse(codeMirror.getValue()), JOB_JSON);
+  });
```

Comment:
Suggestion: add a `percySnapshot` here for a little more reassurance down the road! I can see us making changes to our editor over time and this test would be a nice canary in the coal mine.

Reply:
We're expecting a visual regression where we may change the order of concatenation, as well as, the removal of an alert because of API changes. It might make sense to wait until this issue is resolved and then tackle adding Percy snapshots into another PR. I'm filing a [ticket](https://github.com/hashicorp/nomad/issues/17129) for this.